eth/downloader/peer.go

```diff
@@ -441,6 +483,20 @@ func (ps *peerSet) BlockIdlePeers() ([]*peer, int) {
 	return ps.idlePeers(61, 61, idle, throughput)
 }
 
+// HeaderIdlePeers retrieves a flat list of all the currently header-idle peers
+// within the active peer set, ordered by their reputation.
+func (ps *peerSet) HeaderIdlePeers() ([]*peer, int) {
+	idle := func(p *peer) bool {
+		return atomic.LoadInt32(&p.headerIdle) == 0
+	}
+	throughput := func(p *peer) float64 {
```

Comment:
Can't say the reason for originally doing it like this. This specific version is to be in line with the rest of the method closures, but I don't see why not all of the closures could be separated out as fully qualified methods on the peer. We can do that in a follow up PR if need be, I'd rather not modify all the other code too in this PR.